Sources/MoEngage-Swift-Segment/MoEngage_Swift_Segment.swift

```diff
@@ -0,0 +1,240 @@
+                
+            }
+            let moe_properties = MoEngageProperties(withAttributes: generalAttributeDict)
+            for key in dateAttributeDict.keys {
+                guard let key = key as? String else {
+                    continue
+                }
+                if let dateVal = dateAttributeDict[key] as? Date {
+                    moe_properties.addDateAttribute(dateVal, withName: key)
+                }
+            }
+            MoEngageSDKAnalytics.sharedInstance.trackEvent(event.event, withProperties: moe_properties, forAppID: moengageSettings?.apiKey)
+        }
+
+        return event
+    }
+    
+
+    func date(fromISOdateStr isoDateStr: String?) -> Date? {
+        var dateFormatter: DateFormatter?
+        if let isoDateStr {
+            dateFormatter = DateFormatter()
+            dateFormatter!.locale = Locale(identifier: "en_US_POSIX")
+            dateFormatter!.dateFormat = "yyyy'-'MM'-'dd'T'HH':'mm':'ss.SSS'Z'"
+            dateFormatter!.timeZone = NSTimeZone(forSecondsFromGMT: 0) as TimeZone
+            return dateFormatter?.date(from: isoDateStr)
+        }
+        return nil
+    }
+    
+    public func alias(event: AliasEvent) -> AliasEvent? {
+        let appId = moengageSettings?.apiKey
+        if let userId = event.userId {
+            MoEngageSDKAnalytics.sharedInstance.setAlias(userId, forAppID: appId)
+        }
+        return event
+    }
+    
+    public func flush() {
+        let appID = moengageSettings?.apiKey
+        MoEngageSDKAnalytics.sharedInstance.flush(forAppID: appID)
+    }
+    
+    public func reset() {
+        let appID = moengageSettings?.apiKey
+        MoEngageSDKAnalytics.sharedInstance.resetUser(forAppID: appID)
+    }
+}
+
+// MARK: - Push Notification methods
+
+extension MoEngageDestination: RemoteNotifications {
+    public func registeredForRemoteNotifications(deviceToken: Data) {
+        MoEngageSDKMessaging.sharedInstance.setPushToken(deviceToken)
+    }
+    
+    public func failedToRegisterForRemoteNotification(error: Error?) {
+        MoEngageSDKMessaging.sharedInstance.didFailToRegisterForPush()
+    }
+    
+    public func receivedRemoteNotification(userInfo: [AnyHashable : Any]) {
+        MoEngageSDKMessaging.sharedInstance.didReceieveNotification(withInfo: userInfo)
+    }
+}
+
+// MARK: - User Notification Center delegate methods
+extension MoEngageDestination: UNUserNotificationCenterDelegate {
+    @available(iOS 10.0, *)
+    public func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
+        completionHandler([.alert , .sound])
+    }
+    
+    @available(iOS 10.0, *)
+    public func userNotificationCenter(_ center: UNUserNotificationCenter, didReceive response: UNNotificationResponse, withCompletionHandler completionHandler: @escaping () -> Void) {
+        completionHandler()
```

Comment:
@deepa-iOS-SDK if the delegate is set to this class and as proxy won't work as expected in segment, I assume moengage callback are lost as no methods are called